### PR TITLE
Cover the max code length validation in bundles

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13
| Bug fix?        | kinda?
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php [hunk 1]
```diff
@@ -57,7 +61,7 @@ public function iWantToCreateANewChannel(): void
      * @When I specify its code as :code
      * @When I do not specify its code
      */
-    public function iSpecifyItsCodeAs(string $code = null)
+    public function iSpecifyItsCodeAs(?string $code = null): void
```
Comment (jakubtobiasz) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingChannelsContext.php:64]: All changes like that were required to meet the trait's conctract :)

File: features/addressing/managing_countries/province_validation.feature [hunk 2]
```diff
@@ -17,6 +17,14 @@ Feature: Province validation
         Then I should be notified that code is required
         And province with name "Scotland" should not be added in this country
 
+    @ui @mink:chromedriver @api
+    Scenario: Trying to add a new province with a too long code
+        Given I want to create a new province in country "United Kingdom"
+        And I name the province "Scotland"
+        When I specify a too long province code
```
Comment (GSadee) [inline features/addressing/managing_countries/province_validation.feature]:
```suggestion
        When I want to create a new province in country "United Kingdom"
        And I name the province "Scotland"
        And I specify a too long province code
```
and change that also in other scenarios

File: src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin\Helper;
+
+use Webmozart\Assert\Assert;
+
+trait CodeValidationTrait
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php]: Why didn't you add this as a separate context (`ValidatingCodeContext` or sth similar)?

File: src/Sylius/Behat/Context/Api/Admin/ManagingCountriesContext.php [hunk 4]
```diff
@@ -358,6 +369,21 @@ public function iShouldBeNotifiedThatNameOfTheProvinceIsRequired(): void
...
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'The code must not be longer than',
+        );
+    }
+
+    /**
+     * @Then I shou
+     */
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingCountriesContext.php]:
```suggestion
```

File: src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php (hunk 3 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php]: Every context has different `createPage/updatePage` and that's why I pick the trai way, but when I think about it now, I could create a custom element 🤔.

File: features/addressing/managing_countries/province_validation.feature (hunk 2 above)
Comment (jakubtobiasz) [reply features/addressing/managing_countries/province_validation.feature]: I treat `Given` as a set of steps required (and preceding the place where I'm currently), so the `When` is emphazing the part I want "check" 🤔.

So when you read my scenarios, you can skip `Given` and you know when you try to specify too long code, you'll get an error 🤔.

Comment (jakubtobiasz) [reply features/addressing/managing_countries/province_validation.feature]: Done

File: src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php (hunk 3 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Context/Api/Admin/Helper/CodeValidationTrait.php]: I wasn't able to make a separate context, the easiest way is to access the `create/update` page, I don't see a way to make it possible now. Also, I don't want to spend too much time on it. I've just make the trait more generic.

**Review by Wojdylak — APPROVED** [review]

**Comment by Wojdylak** [comment]
Thank you, @jakubtobiasz!

**Pull request merged**